File: src/main/java/com/epicdragonworld/gameserver/data/NpcData.java
package com.epicdragonworld.gameserver.data;

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.util.HashMap;
import java.util.Map;
import java.util.logging.Logger;

import com.epicdragonworld.gameserver.enums.NpcType;
import com.epicdragonworld.gameserver.managers.DatabaseManager;
import com.epicdragonworld.gameserver.model.holders.NpcHolder;

/**
 * @author Pantelis Andrianakis
 */
public class NpcData
{
	private static final Logger LOGGER = Logger.getLogger(NpcData.class.getName());
	
	private static final String RESTORE_NPCS = "SELECT * FROM npcs";
	
	private static final Map<Integer, NpcHolder> _npcs = new HashMap<>();
	
	public static void init()
	{
		_npcs.clear();
		
		try (Connection con = DatabaseManager.getConnection();
			PreparedStatement ps = con.prepareStatement(RESTORE_NPCS))
		{
			try (ResultSet rset = ps.executeQuery())
			{
				while (rset.next())
				{
					final int npcId = rset.getInt("npc_id");
					_npcs.put(npcId, new NpcHolder(npcId, rset.getInt("level"), Enum.valueOf(NpcType.class, rset.getString("type")), rset.getInt("stamina"), rset.getInt("strength"), rset.getInt("dexterity"), rset.getInt("intelect")));
				}
			}
		}
		catch (Exception e)
		{
			LOGGER.warning(e.getMessage());
		}
		
		LOGGER.info("NpcData: Loaded " + _npcs.size() + " npcs.");
	}
	
	public static NpcHolder getNpcHolder(int npcId)
	{
		return _npcs.get(npcId);
	}
}

File: EpicDragonWorldServer/Data/NpcData.cs
using MySql.Data.MySqlClient;
using System;
using System.Collections.Generic;

/**
 * Author: Pantelis Andrianakis
 * Date: November 28th 2019
 */
public class NpcData
{
    private static readonly string RESTORE_NPCS = "SELECT * FROM npcs";
    private static readonly Dictionary<long, NpcHolder> NPCS = new Dictionary<long, NpcHolder>();

    public static void Load()
    {
        Util.PrintSection("NPCs");

        NPCS.Clear();
        try
        {
            MySqlConnection con = DatabaseManager.GetConnection();
            MySqlCommand cmd = new MySqlCommand(RESTORE_NPCS, con);
            MySqlDataReader reader = cmd.ExecuteReader();
            while (reader.Read())
            {
                int npcId = reader.GetInt32("npc_id");
                NPCS.Add(npcId, new NpcHolder(npcId, (NpcType)Enum.Parse(typeof(NpcType), reader.GetString("type")), reader.GetInt32("level"), reader.GetBoolean("sex"), reader.GetInt32("stamina"), reader.GetInt32("strength"), reader.GetInt32("dexterity"), reader.GetInt32("intelect")));
            }
            con.Close();
        }
        catch (Exception e)
        {
            LogManager.Log(e.ToString());
        }

        LogManager.Log("NpcData: Loaded " + NPCS.Count + " NPCs.");
    }

    public static NpcHolder GetNpcHolder(int npcId)
    {
        if (!NPCS.ContainsKey(npcId))
        {
            return null;
        }
        return NPCS[npcId];
    }
}
